aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,109 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void pre_check_gradient(const Tensor& self, IntArrayRef axis, int64_t spacing_size,  int64_t edge_order) {
+  TORCH_CHECK(self.scalar_type() != ScalarType::Byte, "torch.gradient does not support uint8 as input.");
+  TORCH_CHECK( spacing_size == axis.size(), "torch.gradient should receive axis and spacing size to be equal.");
```

Comment:
"torch.gradient should receive axis and spacing size to be equal." -> "torch.gradent expected its spacing and dim arguments to have the same length, but got a spacing argument of length [spacing length here] and a dim argument of length [dim length here]."